app/controllers/trades_controller.rb

```diff
@@ -0,0 +1,55 @@
+class TradesController < ApplicationController
+  before_action :set_trade, only: [:show, :edit, :update, :cancel]
+  before_action :check_login, only: [:new, :create]
+  def index
+    @open_trades = Trade.open.chronological.includes(:player, :user)
+    @xbox_trades = @open_trades.select { |t| t.user.xbox_user? }
+    @playstation_trades = @open_trades.select { |t| t.user.playstation_user? }
+  end
+
+  def show
+    @user = @trade.user
+    @partner = @trade.partner
+    @player = @trade.player
+    @offers = @trade.offers.includes(:players, :user)
+  end
+
+  def cancel
+    if !logged_in? || current_user.id != @trade.user_id || @trade.status != Trade::STATUSES['open']
```

Comment:
You are using `current_user.id != @trade.user_id` in several places. It might warrant pulling  some of these reused equality checks into their own methods so you can rename them in a way that makes it easier to read what the code is checking as well as get the benefit of reuse too.